ark/type/methods/base.ts

```diff
@@ -140,11 +141,20 @@ interface Type<out t = unknown, $ = {}>
 	optional<r = applyConstraint<t, Optional>>(): instantiateType<r, $>
 
 	default<
-		const value extends this["inferIn"],
+		const value extends Extract<
+			this["inferIn"],
+			Primitive | Primitive[] | Record<string, Primitive>
```

Comment:
Shouldn't this just be `Primitive`? Why would we want to allow object default values?

It also makes things ambiguous, whereas otherwise we can just check if `typeof default` === "function" to know if we should call it.